lit_gpt/tokenizer.py

```diff
@@ -1,19 +1,20 @@
 import json
 from pathlib import Path
 from typing import Optional
-
 import torch
 
 
 class Tokenizer:
     def __init__(self, checkpoint_dir: Path) -> None:
         # some checkpoints have both files, `.model` takes precedence
+        
         if (vocabulary_path := checkpoint_dir / "tokenizer.model").is_file():
             from sentencepiece import SentencePieceProcessor
 
             self.processor = SentencePieceProcessor(model_file=str(vocabulary_path))
             self.backend = "sentencepiece"
             self.bos_id = self.processor.bos_id()
+            self.use_bos = self.bos_id is not None                    
```

Comment:
This code path (for sentencepiece tokenizers) will need to load `tokenizer_config.json` to read https://huggingface.co/meta-llama/Llama-2-7b-hf/blob/main/tokenizer_config.json#L2. I don't think `bos_id is not None` is reliable enough

Reply:
Also remember that these things might not be defined. https://huggingface.co/garage-bAInd/Platypus2-70B-instruct/blob/main/tokenizer_config.json is an example of a sentenepiece config that doesn't include this argument.

Reply:
If not defined I can change it to return false, which should allow the any() to still work. Will push an update to this in a sec